pkg/pool-utils/contracts/RebalancedBasePool.sol

```diff
@@ -0,0 +1,106 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+pragma solidity ^0.7.0;
+pragma experimental ABIEncoderV2;
+
+import "@balancer-labs/v2-asset-manager-utils/contracts/IAssetManager.sol";
+import "@balancer-labs/v2-solidity-utils/contracts/openzeppelin/Address.sol";
+import "@balancer-labs/v2-solidity-utils/contracts/helpers/BalancerErrors.sol";
+
+import "./BasePool.sol";
+import "./interfaces/IRebalancedBasePoolRelayer.sol";
+
+/**
+ * @dev Base Pool associated with a relayer that guarantees it can only be joined/exited from the relayer itself.
+ * This contract as a simple mixin for pools. Implementing pools must make sure to call the BasePool's constructor
+ * properly.
+ */
+abstract contract RebalancedBasePool is BasePool {
```

Comment:
Extremely nitpicky, but I noted we could just inherit from `vault/interfaces/IBasePool`, as we're only overriding interface functions and not relying on `BasePool` at all. 

I don't mind keeping things as-is though, just wanted to point this out.

Reply:
I don't think it is possible cause we need to access `BasePool`'s storage variables here